### PR TITLE
Cleanup workaround

### PR DESCRIPTION
**Pull request by pirj**
It seems that the workaround was exclusively used in the DescriptionExtractor spec.

Am I missing something?

Before submitting the PR make sure the following are checked:

* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [-] Added tests.
* [-] Updated documentation.
* [-] Added an entry to the [changelog](https://github.com/rubocop-hq/rubocop-rspec/blob/master/CHANGELOG.md) if the new code introduces user-observable changes.
* [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).
* [x] ADDRESS FIXME notes

### CODE REVIEW

**Comment by bquorning** [comment]
> It seems that the workaround was exclusively used in the DescriptionExtractor spec.

The `WorkaroundCop` was introduced in #234, and the `DescriptionExtractor` changes were added later on.

The comments say it’s a “Clone of the the normal `RuboCop::Cop::Cop` class so we can rewrite the `inherited` method without breaking functionality”.

That said, I’m also not completely sure whether we need the workaround :-)

**Comment by pirj** [comment]
There's a tiny change TBD here, `RuboCop::Cop::RSpec::Cop` gets documented despite being abstract. I'll clamp this, there's a FIXME flag where this should be done. 

> rewrite the inherited method without breaking functionality

I have no idea what *was* the exact problem there.


```ruby
@registry = Registry.new
class << self
  attr_reader :registry
end

def self.inherited(base)
  registry.enlist(base)
end
```

fails out of the box, because `@registry` is per-class and is `nil` in our `Cop`, so we have to relay `enlist` to the original.

Semi-related: https://github.com/rubocop-hq/rubocop/pull/8023

File: lib/rubocop/rspec/description_extractor.rb [hunk 1]
```diff
@@ -68,11 +68,7 @@ def documented_constant
         end
 
         def cop_subclass?
-          # YARD superclass resolution is a bit flaky: All classes loaded before
-          # RuboCop::Cop::WorkaroundCop are shown as having RuboCop::Cop as
-          # superclass, while all the following classes are listed as having
-          # RuboCop::Cop::RSpec::Cop as their superclass.
```
Comment (pirj) [inline lib/rubocop/rspec/description_extractor.rb:74]: Now it's more or less clear. There's a better way - pass `RuboCop::Cop::RSpec::Cop`'s path as the first argument to `YARD.parse` to avoid its confusion.

**Comment by pirj** [comment]
@bquorning @Darhazer Green!

File: tasks/cops_documentation.rake [hunk 2]
```diff
@@ -172,30 +168,29 @@ task generate_cops_documentation: :yard_for_generate_documentation do
...
 
-  # rubocop:disable Metrics/AbcSize
-  # rubocop:disable Metrics/MethodLength
-  def print_cops_of_department(cops, department, config)
-    selected_cops = cops_of_department(cops, department).select do |cop|
-      cop.to_s.start_with?('RuboCop::Cop::RSpec')
+  def selected_cops(cops, department)
+    cops_of_department(cops, department.to_sym).select do |cop|
+      cop.name.start_with?('RuboCop::Cop::RSpec') &&
+        cop.name != 'RuboCop::Cop::RSpec::Cop'
```
Comment (pirj) [inline tasks/cops_documentation.rake:174]: Skip the base abstract cop from being added to the docs.

File: lib/rubocop/rspec/description_extractor.rb [hunk 3]
```diff
@@ -68,11 +68,7 @@ def documented_constant
...
-          COP_CLASS_NAMES.include?(yardoc.superclass.path)
+          yardoc.superclass.path == COP_CLASS_NAME
```
Comment (pirj) [inline lib/rubocop/rspec/description_extractor.rb:71]: We're passing our base cop's path to YARD first, so it detects the superclass properly, and doesn't erroneously report some of them to be `RuboCop::Cop`.

**Comment by pirj** [comment]
Maybe we should drop `RuboCop::Cop::RSpec::Cop` altogether? The other extensions don't seem to have their own base cops.

**Comment by pirj** [comment]
Ping

**Comment by bquorning** [comment]
FYI I found https://github.com/rubocop-hq/rubocop/pull/3837 which changed the `inherited` hook, which was mentioned in #234 as the reason for `WorkaroundCop`.

File: bin/build_config [hunk 4]
```diff
@@ -9,9 +9,10 @@ require 'rubocop-rspec'
 require 'rubocop/rspec/description_extractor'
 require 'rubocop/rspec/config_formatter'
 
-glob = File.join(__dir__, '..', 'lib', 'rubocop', 'cop', 'rspec',
+glob = File.join('lib', 'rubocop', 'cop', 'rspec',
                  '{,capybara,factory_bot,rails}', '*.rb')
-YARD.parse(Dir[glob], [])
+rspec_cop_path = File.join('lib', 'rubocop', 'cop', 'rspec', 'cop.rb')
+YARD.parse([rspec_cop_path, *Dir[glob]], [])
```
Comment (bquorning) [inline bin/build_config:15]: Since `'{,capybara,factory_bot,rails}', '*.rb')` already covers `cop.rb`, perhaps it should be made more explicit that we _want_ the `cop.rb` entry to be first. Maybe by using `[]#prepend`, or by adding a comment?

File: spec/rubocop/rspec/description_extractor_spec.rb [hunk 5]
```diff
@@ -42,15 +42,24 @@ def bar
...
-    stub_const(
-      "RuboCop::Cop::RSpec::#{name}",
-      Class.new(RuboCop::Cop.const_get(:WorkaroundCop))
-    )
+    stub_const("RuboCop::Cop::RSpec::#{name}", Class.new(temp_class))
   end
 
   before do
     stub_cop_const('Foo')
+    stub_const('RuboCop::Cop::RSpec::Foo::HelperClassForFoo', Class.new)
```
Comment (bquorning) [inline spec/rubocop/rspec/description_extractor_spec.rb:62]: Why is this necessary now?

Comment (pirj) [reply spec/rubocop/rspec/description_extractor_spec.rb]: Nice catch! Something was breaking in the process, but doesn't now.
Removing.

**Comment by bquorning** [comment]
Thank you @pirj . It truly does look like this `Cop` superclass was not necessary anymore.
